src/slice.rs

```diff
@@ -103,55 +112,66 @@ pub enum SliceOrIndex {
     },
     /// A single index.
     Index(isize),
+    /// A new axis of length 1.
+    NewAxis,
 }
 
-copy_and_clone!{SliceOrIndex}
+copy_and_clone!{AxisSliceInfo}
 
-impl SliceOrIndex {
+impl AxisSliceInfo {
     /// Returns `true` if `self` is a `Slice` value.
     pub fn is_slice(&self) -> bool {
         match self {
-            &SliceOrIndex::Slice { .. } => true,
+            &AxisSliceInfo::Slice { .. } => true,
             _ => false,
         }
     }
 
     /// Returns `true` if `self` is an `Index` value.
     pub fn is_index(&self) -> bool {
         match self {
-            &SliceOrIndex::Index(_) => true,
+            &AxisSliceInfo::Index(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Returns `true` if `self` is a `NewAxis` value.
+    pub fn is_new_axis(&self) -> bool {
+        match self {
+            &AxisSliceInfo::NewAxis => true,
```

Comment:
style nit: drop the `&`

(of course hoping to be able to provide some more valuable feedback than that :)